pmtiles/server.go

```diff
@@ -317,8 +317,9 @@ func (server *Server) get_tile(ctx context.Context, http_headers map[string]stri
 
 		if entry.RunLength > 0 {
 			r, err := server.bucket.NewRangeReader(ctx, name+".pmtiles", int64(header.TileDataOffset+entry.Offset), int64(entry.Length))
+			// possible we have the header/directory cached but the archive has disappeared
 			if err != nil {
-				return 500, http_headers, []byte("Network error")
+				return 404, http_headers, []byte("Archive not found")
```

Comment:
Could we pass through the actual error? Or at least log it. Also is archive missing the only kind of failure here? When it does the initial get it can either fail with a 404 or 500:

```go
	if err != nil {
		return 500, http_headers, []byte("I/O Error")
	}
	if !found {
		return 404, http_headers, []byte("Archive not found")
	}
```

Reply:
I don't want to expose internal error messages over HTTP, many servers will be public facing. 

Let me dig more into the failure scenarios where gocloud returns an error

Reply:
Ok that makes sense. Maybe if we just log the error then like if there's an error getting a directory?